beesdoo_shift_swap/wizard/exchange_wizard.py

```diff
@@ -134,13 +138,17 @@ def make_change(self):
         }
         self.env["beesdoo.shift.exchange_request"].sudo().create(data)
 
-    def contact_coop_same_day_same_hour(self):
+    def contact_coop(self):
         """
-        Send a mail to workers that are subscribed to timesolts with same
-        date and hour as self but on another planning
+        Send a mail to all workers of asked_tmpl_dated_ids
+        before creating the exchange request
         """
-        self.ensure_one()
-        matching_workers = self.exchanged_tmpl_dated_id.get_worker_same_day_same_hour()
-        for worker in matching_workers:
-            self.worker_id.send_mail_for_exchange(self.exchanged_tmpl_dated_id, worker)
-        return True
+        self = self._check()
+        if not self.asked_tmpl_dated_ids:
+            raise UserError(_("Please ask for at least one shift"))
+        for tmpl_dated in self.asked_tmpl_dated_ids:
+            for worker in tmpl_dated.template_id.worker_ids:
+                self.worker_id.send_mail_for_exchange(
+                    self.exchanged_tmpl_dated_id, tmpl_dated, worker
+                )
+        self.make_change()
```

Comment:
Could this method have a more precise name ? Like make_exchange ?

Reply:
Done